### PR TITLE
Hosting model topic trimming/selection

### LINKED ISSUE #23145: Review Blazor Hosting Models topic for trimming
Per https://github.com/dotnet/AspNetCore.Docs/pull/23065#discussion_r697629003, review looking for opportunities to shorten and sharpen the guidance for making choice of hosting model. Extra information should be offloaded to other topics where possible.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 542f38c4-1502-f739-372a-80c86c377491
* Version Independent ID: 72c2a844-ffb2-4799-2c7a-a7a73d3b6833
* Content: [ASP.NET Core Blazor hosting models](https://docs.microsoft.com/en-us/aspnet/core/blazor/hosting-models?view=aspnetcore-5.0)
* Content Source: [aspnetcore/blazor/hosting-models.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/hosting-models.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #23145

[Internal Review Topic (Hosting Models topic)](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/hosting-models?view=aspnetcore-3.1&branch=pr-en-us-23200)

### CODE REVIEW

**Comment by danroth27** [comment]
Some suggestions here: https://github.com/dotnet/AspNetCore.Docs/issues/23145#issuecomment-915519138

**Comment by guardrex** [comment]
@danroth27 ... Although I'll need to sleep on this 🛌😴 with final updates on Friday, I think you can see if this is resolving your suggestions at this stage.

... and do you want to keep the section/table that I added at the end of the *Hosting Models* topic? If so, what changes do you want to make to it? If not, I'll 🔪 it on Friday morning.

**Comment by danroth27** [comment]
> do you want to keep the section/table that I added at the end of the Hosting Models topic?

Yup, I like the table as a quick decision matrix. I just need to look it over and make sure it covers all the important decision points.

**Comment by guardrex** [comment]
btw -- I improve the *Host and Deploy* > *Blazor Server* topic here by adjusting the headings so that many of them will show up in the doc's ToC in the sidebar. However, this topic is still on the list to receive the full 🦖 **_Rex Treatment_**&trade; 🦖 on the [UE pass tracking issue](https://github.com/dotnet/AspNetCore.Docs/issues/19286). I'm sure that there are a number of additional improvements that I can make when I reach it (hopefully in 22H1 🤞). For now on this PR, I just focus on the hosting model coverage changes that pertain to that topic.

**Comment by guardrex** [comment]
Dan, ignore that status checker failure. It's a false positive on a section header that has no bearing on cross-links to this topic.

File: aspnetcore/blazor/hosting-models.md [hunk 1]
```diff
@@ -150,6 +80,21 @@ Blazor WebAssembly includes support for trimming unused code from .NET Core fram
 
 ::: moniker-end
 
+## Hosting model selection
+
+Choice of the Blazor hosting model is an early consideration for Blazor app development. The following table shows the primary considerations for selecting the hosting model.
+
+| &nbsp; | Blazor Server | Blazor WebAssembly |
+| --- | :---: | :---: |
+| Wide .NET Core API compatibility              | ✔️ | ❌ |
```
Comment (mkArtakMSFT) [inline aspnetcore/blazor/hosting-models.md:89]: I don't want this to become misleading. If we call it 'Wide' it feels like it's incomplete - it's just most of them.
Instead, if we call it "Complete" the perception will be more than Wide. So, should this be renamed to `Complete .NET Core API compatibility`?

Comment (guardrex) [reply aspnetcore/blazor/hosting-models.md:89]: Sure, I was just going off of some Dan lingo, probably from a blog post. I'll change it to "Complete" on the next commit.

File: aspnetcore/blazor/hosting-models.md [hunk 2]
```diff
@@ -150,6 +80,21 @@ Blazor WebAssembly includes support for trimming unused code from .NET Core fram
...
+| Wide 3rd party .NET tooling support           | ✔️ | ❌ |
+| Small payload size with fast initial download | ✔️ | ❌ |
+| App code secure and private on the server     | ✔️ | ❌&dagger; |
+| Serverless deployment                         | ❌ | ✔️ |
```
Comment (mkArtakMSFT) [inline aspnetcore/blazor/hosting-models.md:93]: I'm intentionally trying to look at this from a newbie's perspective and `Serverless` may be a word they're not familiar with. What if we use a simpler description here:

```suggestion
| Possibility to run apps without a backing server                        | ❌ | ✔️ |
```

Note, that this is just an idea. I will be happy to get a push back on this.

**Review by mkArtakMSFT — APPROVED** [review]
Thanks @guardrex!

File: aspnetcore/blazor/hosting-models.md (hunk 1 above)
Comment (mkArtakMSFT) [reply aspnetcore/blazor/hosting-models.md:89]: @danroth27 I hope you have no objections regarding this.

**Comment by guardrex** [comment]
I'll delay merge until @danroth27 gets back to us. 👁️

**Review by danroth27 — CHANGES_REQUESTED** [review]
Looking good! 

In addition to my commented suggestions, I'd also like to update the following:

- In the 2nd paragraph of the Blazor WebAssembly section, the last sentence seems a bit overly specific. Any Blazor WebAssembly app can access backend or server functionality over the network. This can be done using pretty much any mechansim that the browser supports (web API, gRPC-web, SignalR, etc). Can we broaden this sentence a bit to be more inclusive?
- I'd like the Blazor WebAssembly section to call out the benefits of hosting the app using ASP.NET Core. Hosting in ASP.NET Core means you're now getting a full stack web development experience with .NET, including the ability to share code on the client and server, support for prerendering, and integration with MVC & Razor Pages.
- In the benefits list for Blazor WebAssembly, the first and last bullet points seem to say pretty much the same thing. I think the first bullet should be about offline support, and the last bullet should be able static site hosting.
- The paragraph about AOT really should be broadened to also mention for native dependencies. We provide the .NET WebAssembly build tools to enable AOT compilation, runtime relinking, and native dependencies support.
- In addition to code trimming, we should probably also mention that we precompress the app to further reduce its download size.

Let me know if you have any questions or concerns about any of these suggestions!

**Review by danroth27 — APPROVED** [review]
LGTM. Thanks for your patience on this one!

**Pull request merged**